### ISSUE TITLE
[Bug]: Github Authorization token not included on branches requests

### ISSUE THREAD

**Issue by kurko** [ObsidianPublisher/obsidian-github-publisher#31]
### Issue validation

- [X] My issue is not a question relative to [Material Mkdocs](https://squidfunk.github.io/mkdocs-material/) nor [Mkdocs](https://www.mkdocs.org/) (*[You can open a discussion to get more help!](https://github.com/squidfunk/mkdocs-material/discussions)*)
- [X] I checked the issue to prevent duplicate
- [X] I checked my configurations files and the documentation

### Workflow

Obsidian plugin 💠

### Describe the bug

*tl;dr* the plugin, v4.3.6, requests to `/branches` returns 404, despite token configured correctly.

The dev console's Network tab completely omits the `Authorization` header, so I'm not sure whether the header is really missing or just hidden. I debugged the plugin manually in the console and the code for the request works, so it's a mystery (see more about the debugging at the end).

Here are the sequence of events:

Upon trying to share files (Active or all):

<img width="699" alt="Screen Shot 2022-10-08 at 14 09 35" src="https://user-images.githubusercontent.com/70312/194720260-721f3354-d20f-485c-bf09-abd7dff29690.png">

---

The dev console indicates a `404` error:

<img width="1108" alt="Screen Shot 2022-10-08 at 14 05 44" src="https://user-images.githubusercontent.com/70312/194720265-1f3731b8-7b44-4891-a06e-2d18fbcdba64.png">

---

Note that the Network tab doesn't show any Authorization header being included:

<img width="626" alt="Screen Shot 2022-10-08 at 14 06 57" src="https://user-images.githubusercontent.com/70312/194720297-3742bb6a-64fc-41f9-956d-2034add4e44a.png">

<img width="619" alt="Screen Shot 2022-10-08 at 14 07 04" src="https://user-images.githubusercontent.com/70312/194720298-ea0e9236-e988-46a7-8fe7-eca98fa5d8d7.png">

---

However, if I try it manually from the terminal passing the token, it works:

<img width="990" alt="Screen Shot 2022-10-08 at 14 07 19" src="https://user-images.githubusercontent.com/70312/194720322-02d7e731-52d1-4d78-bc5c-42158d09ea65.png">

### How to reproduce ?

1. install the plugin
2. configure the plugin with repo, username and token
3. create a new doc with a simple frontmatter entry containing `share: true`
4. try to share the file
5. you get the error

### Configuration

```json
{
  "githubRepo": "blog-markdown-files",
  "githubName": "kurko",
  "GhToken": "ghp_[REDACTED]",
  "shareKey": "share",
  "ExcludedFolder": "",
  "fileMenu": false,
  "editorMenu": false,
  "downloadedFolder": "obsidian",
  "folderDefaultName": "",
  "yamlFolderKey": "",
  "rootFolder": "",
  "workflowName": "",
  "embedImage": true,
  "defaultImageFolder": "",
  "autoCleanUp": false,
  "autoCleanUpExcluded": "",
  "folderNote": false,
  "convertWikiLinks": false,
  "convertForGithub": false,
  "subFolder": "",
  "embedNotes": false,
  "copyLink": false,
  "mainLink": "",
  "linkRemover": "",
  "hardBreak": false,
  "logNotice": false,
  "convertDataview": true,
  "useFrontmatterTitle": false,
  "censorText": [],
  "inlineTags": false,
  "dataviewFields": [],
  "excludeDataviewValue": [],
  "metadataFileFields": [],
  "frontmatterTitleKey": "title",
  "shareExternalModified": false
}
```

### OS

MacOS

### Anything else?

tl;dr from the console, it seems to be working well. Just not from the regular menus.

I've debugged a bit and found out that if I call the code manually from the console, it succeeds in calling Github. I debugged and then did `window.octokit = octokit` in order to be able to use it outside the debugger, otherwise we can't use `async` within the `debugger` because the browser complains about it.

<img width="895" alt="Screen Shot 2022-10-08 at 14 51 03" src="https://user-images.githubusercontent.com/70312/194720901-519e59dd-cd0d-476f-89bc-3b196c0b86dd.png">

**Comment by kurko** [comment]
Putting the debugger through each line of `shareOneNote()`, **it now worked** and that is a mystery. By "worked" I mean it published the file, but in some calls to `/branches` it still errored out.

Here's a step by step debugging session. Note that at the end I get a success, but also note in the Network tab, there are _red_ `branches` entries in list.

<img width="568" alt="Screen Shot 2022-10-08 at 14 57 51" src="https://user-images.githubusercontent.com/70312/194721326-39a0881a-009c-49a3-a3b7-78c3e7b59ea7.png">

<img width="561" alt="Screen Shot 2022-10-08 at 14 59 06" src="https://user-images.githubusercontent.com/70312/194721327-14fcfbde-3b00-43d7-8dd2-da99cdb0db08.png">

<img width="780" alt="Screen Shot 2022-10-08 at 14 59 57" src="https://user-images.githubusercontent.com/70312/194721328-0b344e94-820c-48f9-ac38-011f6e6ff1a5.png">

<img width="776" alt="Screen Shot 2022-10-08 at 15 00 44" src="https://user-images.githubusercontent.com/70312/194721329-0209b196-16fe-4796-ae2b-7a72536026f0.png">

**Comment by Lisandra-dev** [comment]
Hello !
Pretty strange bug... 
What is the link to your github repository ? 

I don't have these bug on my side so... It will be difficult to debug

**Comment by kurko** [comment]
@Lisandra-dev [kurko/blog-markdown-files](https://github.com/kurko/blog-markdown-files). Yeah, it seems quite strange that it didn't happen to anyone else. I'll keep using the plugin for now that it worked and report back on any new clue I have. It could very well have been something like a race condition with Obsidian not loading `data.json` properly or something ¯\_(ツ)_/¯

**Comment by Lisandra-dev** [comment]
Publish active will never work for this plugin...

**Comment by Lisandra-dev** [comment]
I finally successfully found the culprit
Normally, this bug is now fixed!

**Comment by kurko** [comment]
Thank you!
